src/extensibility/node/ExtensionManagerDomain.js

```diff
@@ -98,17 +147,40 @@ function _performInstall(packagePath, installDirectory, validationResult, callba
         var sourceDir = path.join(validationResult.extractDir, validationResult.commonPrefix);
 
         fs.copy(sourceDir, installDirectory, function (err) {
-            if (err) {
+
+            var fail = function (err) {
                 _removeFailedInstallation(installDirectory);
                 callback(err, null);
-            } else {
+            };
+
+            var finish = function () {
                 // The status may have already been set previously (as in the
                 // DISABLED case.
                 if (!validationResult.installationStatus) {
                     validationResult.installationStatus = Statuses.INSTALLED;
                 }
                 callback(null, validationResult);
+            };
```

Comment:
Could you move `fail` and `finish` function outside this block?
I think starting the block with the `if (err)` would make think a little bit clearer.